core/src/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -1,40 +1,239 @@
+    // StateMachine type.
+    fn adapt_response(
+        &self,
+        my_command: Self::Command,
+        _event_info: Option<EventInfo>,
+    ) -> Result<Vec<MachineResponse>, Self::Error> {
+        let UpsertSearchAttributesMachineCommand::SendUpsertCommand(attrs) = my_command;
+        Ok(vec![MachineResponse::IssueNewCommand(Command {
+            command_type: CommandType::UpsertWorkflowSearchAttributes as i32,
+            attributes: Some(attrs),
+        })])
+    }
+
+    // Filters for EventType::UpsertWorkflowSearchAttributes
+    fn matches_event(&self, event: &HistoryEvent) -> bool {
+        matches!(
+            event.event_type(),
+            EventType::UpsertWorkflowSearchAttributes
+        )
+    }
+
+    fn kind(&self) -> MachineKind {
+        MachineKind::UpsertSearchAttributes
+    }
+}
+
+impl Cancellable for UpsertSearchAttributesMachine {}
+
+// Converts the generic history event with type EventType::UpsertWorkflowSearchAttributes into the
+// UpsertSearchAttributesMachine-specific event type
+// UpsertSearchAttributesMachineEvents::CommandRecorded.
+impl TryFrom<HistoryEvent> for UpsertSearchAttributesMachineEvents {
+    type Error = WFMachinesError;
 
-impl UpsertCommandCreated {
-    pub(super) fn on_upsert_workflow_search_attributes(
-        self,
-    ) -> UpsertSearchAttributesMachineTransition<UpsertCommandRecorded> {
-        unimplemented!()
+    fn try_from(e: HistoryEvent) -> Result<Self, Self::Error> {
+        match e.event_type() {
+            EventType::UpsertWorkflowSearchAttributes => {
+                Ok(UpsertSearchAttributesMachineEvents::CommandRecorded)
+            }
+            _ => Err(Self::Error::Nondeterminism(format!(
+                "UpsertWorkflowSearchAttributesMachine does not handle {e}"
+            ))),
+        }
     }
 }
 
-#[derive(Default, Clone)]
-pub(super) struct UpsertCommandRecorded {}
+// Converts generic state machine command type CommandType::UpsertWorkflowSearchAttributes into
+// the UpsertSearchAttributesMachine-specific event
+impl TryFrom<CommandType> for UpsertSearchAttributesMachineEvents {
+    type Error = WFMachinesError;
+
+    fn try_from(c: CommandType) -> Result<Self, Self::Error> {
+        match c {
+            CommandType::UpsertWorkflowSearchAttributes => {
+                Ok(UpsertSearchAttributesMachineEvents::CommandScheduled)
+            }
+            _ => Err(Self::Error::Nondeterminism(format!(
+                "UpsertWorkflowSearchAttributesMachine does not handle command type {c:?}"
+            ))),
+        }
+    }
+}
+
+// There is no Command/Reponse associated with this transition
+impl From<CommandIssued> for Done {
+    fn from(_: CommandIssued) -> Self {
+        Self {}
+    }
+}
+
+// There is no Command/Reponse associated with this transition
+impl From<Created> for CommandIssued {
+    fn from(_: Created) -> Self {
+        Self {}
+    }
+}
```

Comment:
I think you can remove these and the state machine impl automatically uses `Default`

Reply:
Doesn't compile without them - maybe I'm missing something somewhere else?